go/chat/convsource.go

```diff
@@ -485,28 +495,76 @@ func (m ByMsgID) Len() int           { return len(m) }
 func (m ByMsgID) Swap(i, j int)      { m[i], m[j] = m[j], m[i] }
 func (m ByMsgID) Less(i, j int) bool { return m[i].GetMessageID() > m[j].GetMessageID() }
 
-func (s *HybridConversationSource) GetMessages(ctx context.Context, convID chat1.ConversationID,
-	uid gregor1.UID, msgIDs []chat1.MessageID, finalizeInfo *chat1.ConversationFinalizeInfo) ([]chat1.MessageUnboxed, error) {
-
-	rmsgsTab := make(map[chat1.MessageID]chat1.MessageUnboxed)
-
-	msgs, err := s.storage.FetchMessages(ctx, convID, uid, msgIDs)
+// getMessagesShared gets messages by ID.
+// The returned messages correspond to `getMsgIDs`.
+// All the new messages in `preFetched` are also saved.
+func (s *HybridConversationSource) getMessagesShared(
```

Comment:
I wonder if this function is too complicated, wouldn't it be easier to just reincarnat the old `GetMessagesWithRemotes`? It just seems like this is trying too hard to make one function when two might be better. Let me know what you think.